### PR TITLE
Update subscription docs

### PR DESCRIPTION
**Pull request by HannesBenson**
## Change description

> Update the documentation related to Topic Subscriptions after the update to the Subscriptions API.

## Type of change

- [ ] Bug (fixes an issue)
- [x] Enhancement (adds functionality)

## Related issues

> Fix MAG-777

### Development

- [x] Lint rules pass locally
- [x] Application changes have been tested thoroughly
- [x] Automated tests covering modified code pass

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached as necessary
- [ ] "Ready for review" label attached and reviewers assigned
- [ ] Changes have been reviewed by at least one other contributor
- [ ] Pull request linked to task tracker where applicable

### CODE REVIEW

File: docs/subscriptions.mdx [hunk 1]
```diff
@@ -38,13 +38,13 @@ The `{ "topic": { "subscribers": true } }` recipient can be used to send notific
 
 ## How this all works
 
-1. When a user subscribes to a topic in your app, a subscription record is created with the status `subscribed`. You can update subscription status using either our [REST API](/docs/rest-api/reference#update-topic-subscription) or [GraphQL API](/docs/graphql-api/reference#update-a-user's-subscription-status-to-a-topic).
+1. When a user subscribes to a topic in your app, a subscription record is created with the status `subscribed`. You can [unsubscribe](/docs/rest-api/reference#unsubscribe-topic-subscription) and [resubscribe](/docs/rest-api/reference#create-topic-subscription) the user using the REST API.
```
Comment (unamashana) [inline docs/subscriptions.mdx:41]: @HannesBenson I believe the app developer needs to call the subscription creation API endpoint?

File: docs/subscriptions.mdx [hunk 2]
```diff
@@ -38,13 +38,13 @@ The `{ "topic": { "subscribers": true } }` recipient can be used to send notific
...
 1. Our backend [creates the subscription](/docs/graphql-api/reference#create-subscription).
```
Comment (unamashana) [inline docs/subscriptions.mdx:42]: Explred link

**Review by unamashana — CHANGES_REQUESTED** [review]

File: docs/subscriptions.mdx (hunk 2 above)
Comment (HannesBenson) [reply docs/subscriptions.mdx]: Removed.

File: docs/subscriptions.mdx (hunk 1 above)
Comment (HannesBenson) [reply docs/subscriptions.mdx]: @unamashana I've reworded this, when you have time, will you please check the wording?

Comment (unamashana) [reply docs/subscriptions.mdx]: Hey @HannesBenson It looks better, but I don't think it communicates that customers need to call the subscribe API to create subscriptions. Are you referring to subscriptions created by sending notifications? Perhaps we need to make this distinction then.

Comment (HannesBenson) [reply docs/subscriptions.mdx]: @unamashana I've updated it again

**Review by unamashana — APPROVED** [review]

**Pull request merged**